### PR TITLE
[TMVA][SOFIE] Nary operators with multidirectional broadcasting

### PR DESCRIPTION
**Pull request by axmat**
## This Pull request:

Development of the nary operators with multi directional broadcasting for TMVA/SOFIE.

## Checklist:

- [x] tested changes locally

### CODE REVIEW

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2204/cxx17, ROOT-ubuntu2004/default, ROOT-ubuntu18.04/nortcxxmod

File: tmva/sofie/inc/TMVA/ROperator_BasicNary.hxx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+struct NaryOperatorTraits<T, EBasicNaryOperator::Mean> {
+   static const std::string Name() {return "Mean";}
+   static std::string Op(const std::string& res, std::vector<std::string>& inputs) {
+      std::stringstream out;
+      out << "\t" << "\t" << res << " = (" << inputs[0];
+      for (size_t i = 1; i < inputs.size(); i++) {
+         out << " + " << inputs[i];
+      }
+      out << ") / ";
+      if constexpr (std::is_same<T, float>::value) {
```
Comment (lmoneta) [inline tmva/sofie/inc/TMVA/ROperator_BasicNary.hxx:57]: Can you please fix for C++14 ? 
See warning produced

**Comment by axmat** [comment]
Done!

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2204/cxx17, ROOT-ubuntu2004/default, ROOT-ubuntu18.04/nortcxxmod

**Review by lmoneta — APPROVED** [review]
LGTM!
Thank you @axmat for the PR and fixing the remaining warnings !

**Pull request merged**